### PR TITLE
feat: add account popup

### PR DESCRIPTION
**Pull request by jiejasonliu**
## summary!
- adds the account popup
  - can close when outside click or escape press
- updates header to show responsive account group 
- account pfp from Gravatar (defaults to unique 8-bit icon like github if not registered)
- new recoil selector (userInfoState) that has a dependency on the authJwt atom
  - whenever the jwt updates, it asynchronously fetches the user info
  - this is why the `log out` button is seen for a second, it's the `React.Suspense` fallback state

---

https://user-images.githubusercontent.com/29434693/191907829-11321d98-2520-4c0b-af57-af6ca57a5e62.mp4

### CODE REVIEW

File: src/state/auth-jwt.ts [hunk 1]
```diff
@@ -11,6 +17,45 @@ export const authJwtState = atom<AuthJwt | undefined>({
...
+  key: 'userInfoStateAsync',
+  get: async ({ get }) => {
+    const authJwt = get(authJwtState);
+
+    if (!isAuthJwt(authJwt)) {
+      return undefined;
+    }
+
+    // we unfortuantely cannot use `useFetchWrapper` since that adds hooks (breaks rule of hooks)
+    // either we replace this selector with effects in a nested component that updates this atom
```
Comment (jiejasonliu) [inline src/state/auth-jwt.ts:40]: i've seen people put a `RecoilEffectListener` component as a sibling of `App` and then that file basically has a component that has an effect that sets up the dep chain.

(e.g. if atomB's result depends on atomA)
```ts
const fetchWrapper = useFetchWrapper();

useEffect(() => {
   (async () => {
     const someResult = await fetchWrapper.get(...);
     setAtomB(someResult);
   })();
}, [atomA])
```

that way, we could use any hook we want bc it lives in a component.
I won't be doing this atm but if growing pains occur due to async selectors/atoms, we can move to this pattern

File: src/hooks/use-key-press.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
```
Comment (derricklee43) [inline src/hooks/use-key-press.ts:1]: This is really dope!! Ima have to add this to the study lesson, haha

File: src/state/auth-jwt.ts (hunk 1 above)
Comment (derricklee43) [reply src/state/auth-jwt.ts:40]: That makes sense :thumbsup:

Comment (derricklee43) [reply src/state/auth-jwt.ts:40]: That's actually pretty cool

**Review by derricklee43 — APPROVED** [review]
LGMT! That dropdown is so slick

File: src/hooks/use-key-press.ts (hunk 2 above)
Comment (jiejasonliu) [reply src/hooks/use-key-press.ts:1]: that would be so cool. it's pretty barebones right now, so depending on what you need, maybe allow it to take an array of keys or an object of keys to callbacks.

Comment (derricklee43) [reply src/hooks/use-key-press.ts:1]: Yeah, an object of keys to callbacks sounds ideal. We can always figure it out later, haha

**Pull request merged**